feat: highlight active navlinks

janvi01 opened the pull request:
<!--   Thank you for your contribution. Before you submit the pull request:
1. Follow our contribution guidelines
2. Test your changes and attach their results to the pull request.
3. Update the relevant documentation.
-->

**Description**

Highlighted the active nav links in the navbar using `useRouter` hook.

Highlight in the context of the bottom border and color when the nav link page is active.

Screenshots -

<img width="1792" alt="Screenshot 2023-02-17 at 4 29 27 PM" src="https://user-images.githubusercontent.com/34344234/219631352-9ca105ab-26cd-4174-9164-1013a601026d.png">

<img width="1792" alt="Screenshot 2023-02-17 at 4 29 19 PM" src="https://user-images.githubusercontent.com/34344234/219631316-bb742a54-1325-4b55-aff2-aef251566a0e.png">


**Related issue(s)**
Fixes #1340

janvi01 committed:
feat: highlight active navlinks

derberg commented:
this PR was shared by Akshat during community meeting. I do not mind having this feature, but we should not add `underline` effect. Enough is to simply preserve the style you see when hover over the link (darker text)

janvi01 commented:
Thanks @derberg , I will update this PR with the suggested changes.

akshatnema reviewed (changes requested):
Hey @janvi01, I saw the changes you made in the PR. My concern is that respective NavItem should be highlighted for every webpage that belonged under that category. Like for example, for every docs page, the `Docs` NavItem should get highlighted.

![Image](https://user-images.githubusercontent.com/76521428/220863917-6f074ee8-e7f1-41c3-b599-5b0e382a5406.png)

janvi01 committed:
update require changes

janvi01 commented:
Hey, I have removed the underlines and added the link activation for every link under that category.

screenshot -

<img width="1792" alt="Screenshot 2023-02-23 at 3 07 44 PM" src="https://user-images.githubusercontent.com/34344234/220870674-89319a7d-4729-4242-80da-97c28d00a0dd.png">

akshatnema reviewed (commented):
![image](https://user-images.githubusercontent.com/76521428/220876481-16c54242-0b4e-418c-aeb2-b368f4947e31.png)

In this case you are getting it wrong. Kindly update your logic to take the first term of the pathname only.

janvi01 commented:
> ![image](https://user-images.githubusercontent.com/76521428/220876481-16c54242-0b4e-418c-aeb2-b368f4947e31.png)
> 
> In this case you are getting it wrong. Kindly update your logic to take the first term of the pathname only.

ah, sorry I didn't think about this. 

Updated the PR again, do we have any edge case for this new logic? Thanks!

janvi01 committed:
update require changes

akshatnema commented on components/navigation/NavItem.js:
```diff
@@ -8,10 +9,11 @@ export default function NavItem ({
     return (
       <Link href={href}>
-        <a target={target} rel="noopener noreferrer" className={`${className} font-body text-base leading-6 font-semibold text-gray-700 hover:text-gray-900 focus:outline-none focus:text-gray-900 transition ease-in-out duration-150`}>
+        <a target={target} rel="noopener noreferrer" className={`${className} font-body text-base leading-6 font-semibold text-gray-700 hover:text-gray-900 focus:outline-none focus:text-gray-900 transition ease-in-out duration-150 ${router.pathname.substring(0,href.length) === href ? "text-black" :"text-gray-700"}`}>
```
```suggestion
        <a target={target} rel="noopener noreferrer" className={`${className} font-body text-base leading-6 font-semibold text-gray-700 hover:text-gray-900 focus:outline-none focus:text-gray-900 transition ease-in-out duration-150 ${router.pathname.startsWith(href) ? "text-black" :"text-gray-700"}`}>
```

akshatnema commented on components/navigation/NavItem.js:
```diff
@@ -22,7 +24,7 @@ export default function NavItem ({
       {href ? (
         <Link href={href}>
-          <a target={target} rel="noopener noreferrer" className={`${className} font-body text-base leading-6 font-semibold text-gray-700 hover:text-gray-900 focus:outline-none focus:text-gray-900 transition ease-in-out duration-150`}>
+          <a target={target} rel="noopener noreferrer" className={`${className} font-body text-base leading-6 font-semibold text-gray-700 hover:text-gray-900 focus:outline-none focus:text-gray-900 transition ease-in-out duration-150 ${router.pathname.substring(0,href.length) === href ? "text-black" :"text-gray-700"}`}>
```
```suggestion
          <a target={target} rel="noopener noreferrer" className={`${className} font-body text-base leading-6 font-semibold text-gray-700 hover:text-gray-900 focus:outline-none focus:text-gray-900 transition ease-in-out duration-150 ${router.pathname.startsWith(href) ? "text-black" :"text-gray-700"}`}>
```

akshatnema reviewed (changes requested):
Make it more simple so that others can also able to understand what we are doing with the if-else conditions.

janvi01 committed:
update require changes

janvi01 commented:
> Make it more simple so that others can also able to understand what we are doing with the if-else conditions.

Are you talking about including if-else separately? How we would include it in inline styling?

akshatnema commented:
> Are you talking about including if-else separately? How we would include it in inline styling?

Nope, I already corrected that in the changes above.

akshatnema committed:
Merge branch 'master' into activeNavLinks

akshatnema reviewed (commented):
![image](https://user-images.githubusercontent.com/76521428/221139296-d9d199d9-8596-4e07-814d-bac666938a68.png)

I can't see the same activity for the `Community` section pages. Can you check where it is failing?

janvi01 commented:
> ![image](https://user-images.githubusercontent.com/76521428/221139296-d9d199d9-8596-4e07-814d-bac666938a68.png)
> 
> I can't see the same activity for the `Community` section pages. Can you check where it is failing?

Since we don't have a particular `href = '/community'` page when clicking on the community as of now.

akshatnema commented:
> Since we don't have a particular href = '/community' page when clicking on the community as of now

Sorry, I think you didn't get the point I want say. 

In the image if you carefully see, I'm on the Newsroom page i.e. on the route `/community/newsroom`. While on this page, the Community menu text should get highlighted, but isn't working. Kindly check of you have missed any case somewhere.

janvi01 commented:
That's what I am saying. We are doing `router.pathname.startsWith(href)` for each nav item. When we are on route `/community/newsroom` we have `router.pathname` as `/community/newsroom` but we don't have any value of `href = '/community'` as we haven't passed any href value for `community` nav item in `NavBar.js` [here](https://github.com/asyncapi/website/blob/master/components/navigation/NavBar.js#L108).

akshatnema committed:
community error resolved

akshatnema committed:
Merge branch 'master' into activeNavLinks

akshatnema commented:
/rtm
